src/arch/x86_64/CodeGen.zig

```diff
@@ -5483,15 +5483,16 @@ fn airCmp(self: *Self, inst: Air.Inst.Index, op: math.CompareOperator) !void {
         // Source operand can be an immediate, 8 bits or 32 bits.
         // TODO look into reusing the operand
         const lhs = try self.resolveInst(bin_op.lhs);
-        const lhs_lock: ?RegisterLock = switch (lhs) {
+
+        const dst_reg_lock: RegisterLock = switch (lhs) {
             .register => |reg| self.register_manager.lockRegAssumeUnused(reg),
-            else => null,
+            else => blk: {
+                const tmp_reg = try self.copyToTmpRegister(ty, lhs);
+                break :blk self.register_manager.lockRegAssumeUnused(tmp_reg);
+            },
         };
-        defer if (lhs_lock) |lock| self.register_manager.unlockReg(lock);
-
-        const dst_reg = try self.copyToTmpRegister(ty, lhs);
-        const dst_reg_lock = self.register_manager.lockRegAssumeUnused(dst_reg);
         defer self.register_manager.unlockReg(dst_reg_lock);
+        const dst_reg = dst_reg_lock.register;
```

Comment:
Pulling the register out of the lock feels a bit goofy to me, but it was the cleanest way I could come up with. I'd love suggestions for improvements.